tools/autograd/templates/Functions.cpp

```diff
@@ -1671,7 +1673,8 @@ Tensor slogdet_backward(const std::vector<torch::autograd::Variable> &grads,
     return grads[1] * self.inverse().t();
   } else /* otherwise det = \prod(sigma) = 0, use svd */ {
     Tensor u, sigma, v;
-    std::tie(u, sigma, v) = self.svd();
+    bool self_need_grad = self.is_variable() && self.requires_grad();
+    std::tie(u, sigma, v) = self.svd(/*some=*/true, /*compute_uv=*/self_need_grad);
```

Comment:
This is really confusing for me. You’re computing those values only conditionally, but you’re always using them below??

Reply:
omg. solid point. These should always compute u and v. I'll revert these changes. Thanks!